include/envoy/network/address.h

```diff
@@ -178,9 +182,9 @@ class Instance {
   virtual Type type() const PURE;
 
   /**
-   * @return name of socket interface that should be used with this address
+   * @return SocketInterface to be used with the address
    */
-  virtual const std::string& socketInterface() const PURE;
+  virtual const Network::SocketInterface* socketInterface() const PURE;
```

Comment:
let's return reference, it's no longer nullable.

Reply:
We'll have to keep the const because of how we pass the SocketInterface pointer to the Address. Is that okay or were you thinking about something more complex?

Reply:
ah of couse, const wherever possible :)